test/lit/passes/type-refining-gufa.wast

```diff
@@ -296,3 +296,90 @@
   )
 )
 
+;; GUFA can infer the first global contains a $func, so the struct type's field
+;; can be refined. However, doing so would require adding a cast from the
+;; global's declared type (ref func) to the refined type, so that the struct.new
+;; validates. (Alternatively, we would need to refine the global's type at the
+;; same time we refine the struct, but this pass only refines structs.) The type
```

Comment:
_So far_ the pass only refines structs. Would there be any problems in practice if it also refined globals in GUFA mode?

Reply:
It could. It just feels odd for TypeRefining* to do that, but if we move this into the GUFA pass that could make sense.

In any case, these refinements are trivial - GlobalRefining does them already. That is, if GlobalRefining were run before this pass, we wouldn't have the problem. So doing more work here will not result in more total optimizations.